http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java

```diff
@@ -399,6 +411,19 @@ private void closeEventLoopUninterruptibly(EventLoopGroup eventLoopGroup) throws
          * @return This builder for method chaining.
          */
         Builder maxHttp2Streams(Integer maxHttp2Streams);
+
+        /**
+         * Sets the {@link SslProvider} to be used in the Netty client.
+         *
+         * <p>If not configured, {@link SslContext#defaultClientProvider()} will be used to determine the SslProvider.
+         *
+         * <p>Note that you might need to add other dependencies if not using JDK's default Ssl Provider.
+         * See https://netty.io/wiki/requirements-for-4.x.html#transport-security-tls
+         *
+         * @param sslProvider the SslProvider
+         * @return the builder of the method chaining.
+         */
+        Builder sslProvider(SslProvider sslProvider);
```

Comment:
What happens if the provider is not available (missing dependency)?

Reply:
Exception would be thrown when it's sending out the request, for example, `java.lang.UnsatisfiedLinkError: failed to load the required native library` if missing openssl dependency